drivers/hmis/app/graphql/types/hmis_schema/client.rb

```diff
@@ -203,6 +203,9 @@ def enrollments(**args)
     end
 
     def active_enrollment(project_id:, open_on_date:)
+      project = Hmis::Hud::Project.find(project_id)
+      return nil unless current_user.can_view_enrollment_details_for?(project) || current_user.can_view_limited_enrollment_details_for?(project)
```

Comment:
This doesn't quite match the viewable_by scope because viewable_by checks for `(can_view_enrollment_details && can_view_project) || can_view_limited_enrollment_details`. The `Types::HmisSchema::Enrollment` object authorization checks for the same thing. This is definitely a case where it would be better to encapsulate it in an auth policy.

I think with what you have here, it could incorrectly try to resolve an enrollment if you have `can_view_enrollment_details` without `can_view_project` (though it would proceed to raise an exception from the object-level authorization).

Reply:
Thanks for the catch, Gig. If I'm reading this correctly, the former code for `load_open_enrollment_for_client` used the `viewable_by(current_user)` scope on Enrollments _without_ passing `include_limited_access_enrollments`, which is `false` by default. I _think_ we can simplify here to remove the `can_view_limited_enrollment_details` check and that will keep the previous functionality. Does that make sense?

Reply:
Ok, yes, that makes sense! The logic you have looks correct however it causes n+1. `activeEnrollment` is loaded in batch on clients on the bed night screen, so we should resolve that:


```
      project = Hmis::Hud::Project.viewable_by(current_user).find_by(id: project_id)
      return nil unless project.present? && current_user.can_view_enrollment_details_for?(project)

```
->
```
project = load_ar_scope(scope: Hmis::Hud::Project, id: project_id)
return unless current_permission?(permission: :can_view_project, entity: project)
 && current_permission?(permission: :can_view_enrollment_details, entity: project)
```